go/adbc/adbc.go

```diff
@@ -130,6 +131,16 @@ type Error struct {
 	Details []ErrorDetail
 }
 
+// AdbcError helper function returns an adbc.Error representation
+// of a given error
+func AdbcError(err error) (*Error, bool) {
+	var adbcErr Error
+	if errors.As(err, &adbcErr) {
+		return &adbcErr, true
+	}
+	return nil, false
+}
```

Comment:
I don't find this particular function very useful. Consumers should just use `errors.As` and `errors.Is` themselves rather than needing to use this function directly

Reply:
Fine, `error.Is` it is

Reply:
Dropped the new function.